lib/pbench/server/api/resources/datasets_inventory.py

```diff
@@ -63,22 +64,13 @@ def _get(self, params: ApiParams, _) -> Response:
         except TarballNotFound as e:
             raise APIAbort(HTTPStatus.NOT_FOUND, str(e))
 
-        if target is None:
-            file_path = tarball.tarball_path
-        else:
-            dataset_location = tarball.unpacked
-            if dataset_location is None:
-                raise APIAbort(HTTPStatus.NOT_FOUND, "The dataset is not unpacked")
-            file_path = dataset_location / target
+        tarball_name = tarball.tarball_path.name
+        target_path = (
+            tarball_name if target is None else (os.path.join(tarball_name, target))
+        )
+        try:
+            file_path = Tarball.extract(str(tarball.tarball_path), target_path)
+        except Exception as exc:
+            raise APIAbort(HTTPStatus.NOT_FOUND, str(exc))
 
-        if file_path.is_file():
-            return send_file(file_path)
-        elif file_path.exists():
-            raise APIAbort(
-                HTTPStatus.UNSUPPORTED_MEDIA_TYPE,
-                "The specified path does not refer to a regular file",
```

Comment:
I am a bit behind on this one but why are we removing `UNSUPPORTED_MEDIA_TYPE` and return every error as 404 now?

Reply:
Tarball.extract method was handling all error's around this. So, I didn't find the need to include these lines.

Reply:
The cache manager raises an internal error, but we don't want that to fall through to the generic (internal error) `except Exception` in `_dispatch`. That's why we have `APIAbort`, which is treated specially. We want to make sure that the referenced file path exists in the tarball and is a regular file. I'm not sure exactly what `tarfile.extractfile` will return/raise in this case, but we need to specifically catch that case and raise `APIAbort` with `UNSUPPORTED_MEDIA_TYPE`.